backend/app/routers/generate.py

```
@@ -1,204 +1,327 @@
+                    "status": "error",
+                    "error": "Rate limit exceeded. Please try again later.",
+                    "error_code": "RATE_LIMIT",
+                }
+            )
+            log_event(
+                "generate.stream.rate_limited",
+                username=body.username,
+                repo=body.repo,
+                elapsed_ms=timer.elapsed_ms(),
+            )
+        except Exception as exc:
```

Ruff (0.15.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)